### DISCUSSION TITLE
Neorg `sync-parsers` commands fails with NuShell

### DISCUSSION THREAD

**Discussion by JustBarnt** [nvim-neorg/neorg#1547] [Q&A]
Hello, I was recently updating my Neorg config from V7.0.0 to the latest. Which all went smoothly until I went to run: `Neorg sync-parsers` and I was met with this funky error message.

Mainly starting here because I have yet to run into any issues with `nvim-treesitter` itself when install other parsers.

I can resolve it by just running `:set shell=cmd.exe` before I run `:Neorg sync-parsers`

I am just curious if anyone else uses NuShell and has experienced this and made it work without having to change the shell.

```
^[[31mnu::shell::file_not_found^[[0m
  ^[[31m×^[[0m File not found
   ╭─[^[[36;1;4msource^[[0m:1:1]
 ^[[2m1^[[0m │ nu
   · ^[[35;1m▲^[[0m
   · ^[[35;1m╰── ^[[35;1mCould not access file '/s': The system cannot find the file specified. (os error 2)^[[0m^[[0m
   ╰────
{
  cmd = "cmd",
  opts = {
    args = { "/C", "if", "exist", "C:\\Users\\bwilliams\\AppData\\Local\\nvim-data\\tree-sitter-norg", "rmdir", "/s", "/q", "C:\\Users\\bwilliams\\AppData\\Local\\nvim-data\\tree-sitter-norg" }
  }
}
Error: ^[[31mnu::shell::file_not_found^[[0m
  ^[[31m×^[[0m File not found
   ╭─[^[[36;1;4msource^[[0m:1:1]
 ^[[2m1^[[0m │ nu
   · ^[[35;1m▲^[[0m
   · ^[[35;1m╰── ^[[35;1mCould not access file '/s': The system cannot find the file specified. (os error 2)^[[0m^[[0m
   ╰────
{
  cmd = "cmd",
  opts = {
    args = { "/C", "if", "exist", "C:\\Users\\bwilliams\\AppData\\Local\\nvim-data\\tree-sitter-norg_meta", "rmdir", "/s", "/q", "C:\\Users\\bwilliams\\AppData\\Local\\nvim-data\\tree-sitter-norg_meta" }
  }
}
```

**Comment by Or1g3n** [comment]
I have this exact same issue when using nushell in nvim with Jupytext which concerts notebook files to .py and then back after editing. Oddly enough, this issue only impacts one of my machines. On the impacted machine changing the shell to pwsh allows jupytext to work...

**Reply by Or1g3n** [reply]
@JustBarnt 

Eureka!! After many hours of research and trial and error on my own use case, I believe I have found a solution that may work for you:

Try adding the below to your config. I used to only have the first line to specify nu as the shell in neovim. Adding the second line, -c, got me closer to where the error I was getting changed. The 3rd and 4th were the keys that got everything working. At least in my case, the Jupytext plugin was using the vim.fn.shellescape function which in Windows was wrapping file paths in double-quotes; this is problematic in nushell. While nushell does handle file paths with either / or \\, if the file path is wrapped, then nushell treats it like a string, in which case, the \ must be properly escaped, so C:\\\\my\\\\path\\\\to\\\\file. 

So the shellxquote set to empty string prevents the wrapping, and shellslash = true forced forward slash which nushell supports. it will be interesting if this has negative downstream effects BUT at least for now I have not seen anything and my issue is entirely resolved.

``` lua
vim.o.shell = "nu.exe"
vim.o.shellcmdflag = "-c" --Ensures that NuShell runs commands in the proper context
vim.o.shellxquote = ""  -- Prevents extra quoting issues
vim.o.shellslash = true -- Forces Neovim to use forward slashes
```
